sensors/sensor_mock.go

```diff
@@ -2,15 +2,31 @@ package sensors
 
 import (
 	"context"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/spatialmath"
 )
 
-// TimedSensorMock represents a fake TimedSensor.
-type TimedSensorMock struct {
-	TimedSensorReadingFunc func(ctx context.Context) (TimedSensorReadingResponse, error)
+// TimedLidarSensorMock represents a fake TimedLidarSensor.
+type TimedLidarSensorMock struct {
+	TimedLidarSensorReadingFunc func(ctx context.Context) (TimedLidarSensorReadingResponse, error)
+}
+
+// TimedIMUSensorMock represents a fake TimedIMUSensor.
+type TimedIMUSensorMock struct {
+	TimedIMUSensorReadingFunc  func(ctx context.Context) (TimedIMUSensorReadingResponse, error)
+	MockLinearAccelerationData []r3.Vector
+	MockAngularVelocityData    []spatialmath.AngularVelocity
+}
+
+// TimedLidarSensorReading returns a fake TimedLidarSensorReadingResponse or an error
+// panics if TimedLidarSensorReadingFunc is nil.
+func (tsm *TimedLidarSensorMock) TimedLidarSensorReading(ctx context.Context) (TimedLidarSensorReadingResponse, error) {
+	return tsm.TimedLidarSensorReadingFunc(ctx)
```

Comment:
Great question, no idea - the only thing I changed about the code was renaming from TimedSensorReading to TimedLidarSensorReading, so that was like that before. I also used the same structure for IMU. My understanding is that TimedLidarSensorReadingFunc can return an error, though I am not sure about it being unimplemented.

Reply:
@nicksanford Is there a reason to panic here vs returning an error? In RDK this type of mock/injecting usually leads to an error

Reply:
The intention was to communicate that it is a developer error if `TimedLidarSensorReadingFunc` isn't defined to prevent developers from not defining it to test an error path. 

It would be ok to change it to always return an error if it wasn't mocked properly if that would be more in line with RDK patterns & more intuitive.

Reply:
How do you want me to proceed @jeremyrhyde?

Reply:
Ignore for now